### PR TITLE
Fixing routes import, extending local-credentials and adding more routes from Michal Klos

### PR DESCRIPTION
**Pull request by Sinnohd**
Routes import is missing in server
Local-credentials extended so that latest php-sdk works with this
Adding the more routes patch from Michal Klos

### CODE REVIEW

File: metadata/routes.py [hunk 1]
```diff
@@ -5,6 +5,44 @@
 from metadata.bottle import route, response, view, delete, post, request
 
 
+@route('/latest/meta-data/local-hostname')
+def localhost():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'localhost'
+
+@route('/latest/meta-data/hostname')
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E302 expected 2 blank lines, found 1`

File: metadata/routes.py [hunk 2]
```diff
@@ -5,6 +5,44 @@
...
+def host():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'localhost'
+
+@route('/latest/meta-data/public-hostname')
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E302 expected 2 blank lines, found 1`

File: metadata/routes.py [hunk 3]
```diff
@@ -5,6 +5,44 @@
...
+def public_host():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'localhost'
+
+@route('/latest/meta-data/mac')
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E302 expected 2 blank lines, found 1`

File: metadata/routes.py [hunk 4]
```diff
@@ -5,6 +5,44 @@
...
+def vpc_id():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'mac1234'
+
+@route('/latest/meta-data/network/interfaces/macs/mac1234/vpc-id')
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E302 expected 2 blank lines, found 1`

File: metadata/routes.py [hunk 5]
```diff
@@ -5,6 +5,44 @@
...
+def vpc_id():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'vpc-1234'
+
+@route('/latest/meta-data/local-ipv4')
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E302 expected 2 blank lines, found 1`

File: metadata/routes.py [hunk 6]
```diff
@@ -5,6 +5,44 @@
...
+def localip4():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return '127.0.1.1'
+
+@route('/latest/meta-data/instance-type')
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E302 expected 2 blank lines, found 1`

File: metadata/routes.py [hunk 7]
```diff
@@ -5,6 +5,44 @@
...
+def instance_type():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'r3.2xlarge'
+
+
+
+
 @route('/latest/meta-data/iam/security-credentials/')
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E303 too many blank lines (4)`

File: metadata/routes.py [hunk 8]
```diff
@@ -17,6 +55,7 @@ def get_credentials():
         session = request.app.config.meta_get('metadata', 'obj').get_session()
 
         return {
+             'Code':           'Success',
             'AccessKeyId':     session.access_key,
```
Comment (dump247) [inline metadata/routes.py]: pep8: `E131 continuation line unaligned for hanging indent`

File: bin/server [hunk 9]
```diff
@@ -11,7 +11,7 @@ APP_DIR = path.abspath(path.join(path.dirname(__file__), '..'))
 DEFAULT_CONFIG_FILE = path.relpath(path.join(APP_DIR, 'server.conf'))
 sys.path.append(APP_DIR)
 
-from metadata import bottle, Metadata, Profile
+from metadata import bottle, routes, Metadata, Profile
```
Comment (dump247) [inline bin/server]: routes is imported, but not used

File: metadata/routes.py [hunk 10]
```diff
@@ -5,6 +5,44 @@
...
+
+@route('/latest/meta-data/network/interfaces/macs/mac1234/vpc-id')
+def vpc_id():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'vpc-1234'
+
+@route('/latest/meta-data/local-ipv4')
+def localip4():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return '127.0.1.1'
```
Comment (dump247) [inline metadata/routes.py]: I think you mean `127.0.0.1`

File: metadata/routes.py [hunk 11]
```diff
@@ -5,6 +5,44 @@
...
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'localhost'
+
+@route('/latest/meta-data/public-hostname')
+def public_host():
+    response.content_type = 'text/plain; charset=UTF-8'
+    return 'localhost'
+
+@route('/latest/meta-data/mac')
+def vpc_id():
```
Comment (dump247) [inline metadata/routes.py]: Duplicate method name

**Comment by dump247** [comment]
@Sinnohd what is the motivation for adding these new routes?

File: metadata/routes.py (hunk 10 above)
Comment (Sinnohd) [inline metadata/routes.py]: It doesn't matter as long as it is part of 127/8

**Comment by Sinnohd** [comment]
The motivation is, to increase the compatibility with the service running on the AWS instances.

**Pull request merged**